packages/player/src/components/hls-source.tsx

```diff
@@ -25,66 +25,66 @@ export const HLSSource: React.FC<HLSSourceProps> = ({
   type = 'application/x-mpegURL',
   hlsConfig,
 }) => {
-  const [hlsSupported, sethlsSupported] = React.useState(true)
-
   const videoService = useVideo()
   const video = useSelector(videoService, selectVideo)
 
   React.useEffect(() => {
     let hls: Hls
 
+    if (!video) return
+
     function _initPlayer() {
-      if (hls != null) {
+      if (hls) {
         hls.destroy()
       }
 
-      const newHls = new Hls({
+      hls = new Hls({
         enableWorker: false,
         ...hlsConfig,
       })
 
       if (video) {
-        newHls.attachMedia(video)
+        hls.attachMedia(video)
       }
 
-      newHls.on(Hls.Events.MEDIA_ATTACHED, () => {
-        newHls.loadSource(src)
+      hls.on(Hls.Events.MEDIA_ATTACHED, () => {
+        hls.loadSource(src)
       })
 
-      newHls.on(Hls.Events.ERROR, function (event, data) {
+      hls.on(Hls.Events.ERROR, function (event, data) {
         if (data.fatal) {
           switch (data.type) {
             case Hls.ErrorTypes.NETWORK_ERROR:
-              newHls.startLoad()
+              hls.startLoad()
               break
             case Hls.ErrorTypes.MEDIA_ERROR:
-              newHls.recoverMediaError()
+              hls.recoverMediaError()
               break
             default:
               _initPlayer()
               break
           }
         }
       })
-
-      hls = newHls
     }
 
+    const canUseNative = video.canPlayType('application/vnd.apple.mpegurl')
+    const shouldUseNative = canUseNative && !Hls.isSupported()
+
     // Check for Media Source support
-    if (Hls.isSupported()) {
-      sethlsSupported(true)
+    if (!shouldUseNative) {
       _initPlayer()
     } else {
-      sethlsSupported(false)
+      video.load()
```

Comment:
on native players (not using hls.js) we want to tell the element to load and this causes the dynamic sources to be recognized. 

assumed that hls.js is doing this under the hood when it does it's things on other browsers